criu/memfd.c

```diff
@@ -342,6 +342,9 @@ int memfd_open(struct file_desc *d, u32 *fdflags)
 	/* Reopen the fd with original permissions */
 	flags = fdflags ? *fdflags : mfe->flags;
 
+	if (filemap && (flags & O_ACCMODE) == O_RDWR)
```

Comment:
Isn't `filemap == !fdflags` always true?

Reply:
No, it isn't:
https://github.com/checkpoint-restore/criu/blob/5e37ccf379ad7f47a720f64bced40cf2fe5bd6a2/criu/files-reg.c#L2512